File: frontend/check_vue.py
import os
import re

def check_files(start_dir):
    for root, dirs, files in os.walk(start_dir):
        for file in files:
            if file.endswith('.vue'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                    scripts = re.findall(r'<script', content)
                    if len(scripts) > 1:
                        print(f"MULTIPLE SCRIPTS: {path}")

check_files('src')

File: frontend-vue-3/check_vue.js
const fs = require('fs')
const path = require('path')

function walk(dir) {
  const files = fs.readdirSync(dir)
  files.forEach(file => {
    const fullPath = path.join(dir, file)
    if (fs.statSync(fullPath).isDirectory()) {
      walk(fullPath)
    } else if (file.endsWith('.vue')) {
      const content = fs.readFileSync(fullPath, 'utf8')
      const scripts = content.match(/<script/g)
      if (scripts && scripts.length > 1) {
        console.log(`MULTIPLE SCRIPTS: ${fullPath}`)
      }
    }
  })
}

walk('src')
